src/copy.c

```diff
@@ -193,6 +199,12 @@ TSCopyMultiInsertBufferInit(Point *point)
 	buffer->point = palloc(POINT_SIZE(point->num_coords));
 	memcpy(buffer->point, point, POINT_SIZE(point->num_coords));
 
+	/*
+	 * Make a non-refcounted copy of tupdesc to avoid spending CPU in
+	 * ResourceOwner when creating a big number of table slots.
+	 */
+	buffer->tupdesc = CreateTupleDescCopyConstr(cis->rel->rd_att);
```

Comment:
Wouldn't it make sense to disable the reference counting for our local copy of the `tupledesc`? In my benchmarks, this results in a slight performance increase.

```suggestion
	buffer->tupdesc = CreateTupleDescCopyConstr(cis->rel->rd_att);
	/* Disable reference counting */
	buffer->tupdesc->tdrefcount = -1;
```

Reply:
It should be already disabled for the copy, I'll add an assertion.

Reply:
Yes, you are right. I like the idea of adding an assertion to clearly indicate the disabled reference counting.